### PR TITLE
Answers to assignments

### PR DESCRIPTION
**Pull request by chandankj399**

### CODE REVIEW

File: answers.txt [hunk 1]
```diff
@@ -1,19 +1,19 @@
 1. What is a gitignore file and what should it include for a Java project?
 
-A. gitingnore file consists of lists of extensions which the repository should not contain.
-   For example, for a java project we do not want .class, .project or .jar and .war files to be included in our repoitory. We would need only .java files mainly for our source control.
```
Comment (Nagarjun1504) [inline answers.txt]: Java projects create various kinds of extension based on the IDE as well. Can you please name some files which are to be ignored for Eclipse and IntelliJ?

File: answers.txt [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-   In that case, the gitignore file will have .calss, .project, .jar etc as its contents
+A. gitingnore file consists of lists of extensions which the repository should not contain
+   For example, for a java project we do not want .class, .project or .jar and .war files to be included in our repoitory. We would need only .java files mainly for our source control
```
Comment (Nagarjun1504) [inline answers.txt:4]: Java projects create various kinds of extension based on the IDE as well. Can you please name some files which are to be ignored for Eclipse and IntelliJ?

File: answers.txt [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-3. Why should you have different branches?
+3. Why should you have different branches ?
  
-   There are multiple reasons for having branches
-   - When you don't want to mess up your working code of master branch
-   - When multiple people are working on multiple features, create branch for each feature. Once it is completed merge with master and continue
-   - Code reviews will become easier since only a chunk of code of feature branch will have to be reviewed.
+   There are multiple reasons for having branches:
+   - When you don't want to mess up your working code of master branch.
+   - When multiple people are working on multiple features, create branch for each feature. Once it is completed merge with master and continue.
```
Comment (Nagarjun1504) [inline answers.txt:28]: Is it only for new features? What there is a bug or a hotfix required in production environment? We should create a new branch to rectify those bugs as well right?

File: answers.txt [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+   - Code reviews will become easier since only a chunk of code of feature branch will have to be reviewed
 
 4. In a project how many remote Git repositories do you usually have?
    Just one git repository for a project
```
Comment (Nagarjun1504) [inline answers.txt:33]: Is there any scenario wherein we need more than one repository?

File: answers.txt (hunk 2 above)
Comment (chandankj399) [reply answers.txt:4]: For IntelliJ, below file extensions can be ignored
.idea/modules.xml
.idea/*.iml
.idea/modules
*.iml
*.ipr

For Eclipse, below file extensions can be ignored
.settings/
.classpath
.project

File: answers.txt (hunk 3 above)
Comment (chandankj399) [reply answers.txt:28]: Creating a new branch to fix the specific bug will be really helpful. Will add this to the answers file

File: answers.txt (hunk 4 above)
Comment (chandankj399) [reply answers.txt:33]: One scenario I can think of is when you want to create read only replica of your project and share it with other people, you can have read-only repo and your main repository can be read-write.

**Review by Nagarjun1504 — APPROVED** [review]

**Pull request merged**